### ISSUE TITLE
Not sure why the spread function is not working 

### ISSUE THREAD

**Issue by jasminacosta** [Watts-College/cpp-529-spr-2023#17]
Hi @AntJam-Howell,

I am trying to transform the data for Lab #2 but it seems that my code continuously gets stuck at the spread function and I am unsure why. 

median_home_and_income <- c( median_household_income = "B21004_001", 
              median_home_value = "B25077_001")

CenDF <- get_acs(geography= "county", year = 2017, survey= "acs5",
    variables= median_home_and_income, geometry=TRUE)

CenDF <- CenDF %>% 
  mutate(variable= case_when( 
   variable== "B21004_001" ~ "HHIncome",
   variable== "B25077_001" ~ "HouseValue")) %>% 
  select(-moe) %>% 
  spread(variable, estimate) %>%
  mutate(HHInc_HousePrice_Ratio = HouseValue / HHIncome)

Backtrace:
 1. ... %>% ...
 4. sf:::spread.sf(., variable, estimate)
 7. tidyr:::spread.data.frame(...)
Error in tidyr::spread(data, !!key, !!value, fill = fill, convert = convert, :

Any help would be appreciated, thank you!

**Comment by AntJam-Howell** [comment]
@jasminacosta The code looks as it should.  It could be the variables you are using from census.  Try with the ones used below: 

`# `B19049_001` corresponds to the total median household income and `B25077_001` corresponds to the median household value.

vars.to.use <- c( HHIncome = "B19049_001",
                  HouseValue = "B25077_001" )

CenDF <- get_acs( geography="county", 
                  year=2017, 
                  survey="acs5",
                  variables=vars.to.use,
                  geometry=TRUE,
                  shift_geo=TRUE )


CenDF<-CenDF %>% 
  select( -moe ) %>%  
  spread( variable, estimate ) %>%
  mutate( HHInc_HousePrice_Ratio = ( HouseValue/HHIncome ) )
`

**Comment by jasminacosta** [comment]
Hi @AntJam-Howell,

The problem was the variables used and it works now! 

I have a question about how the variable B19049_001 was chosen. 
When I used the code:

medincome <-  VarSearch %>%
  mutate( contains.medincome = grepl( "MEDIAN HOUSEHOLD INCOME", label))%>%
  filter( contains.medincome )

To search for the total median household income I got a table of 21 different variables. I noticed many of the labels and concept descriptions were similar, so in this case do we pick such variables in this situation based on geography? 

The B19049_001 geography wise is a block group, but other variables too that are similar to it that seem to have the same label and concept description. 

![Screenshot 2023-10-18 at 8 41 28 AM](https://github.com/Watts-College/cpp-529-spr-2023/assets/136217967/0f4368fe-ddbe-4626-b072-c11061c9dd35)

I am just confused on this part of the process, and would love some clarity.

Thank you!
